### PR TITLE
Order Dish Added Dao Dao impl and model

### PR DESCRIPTION
**Pull request by junzy**
Adds for order_dish
- Model
- DAO
- DAO Impl

### CODE REVIEW

File: src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	private SessionFactory sessionFactory;
+	Session session;
+
+    public OrderDishDAOImpl(Session session) {
+        this.session = session;
+    }
+    
+    // saves orderdish object to db
+	public void save(OrderDish c) {
+		Session session = this.sessionFactory.openSession();
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java]: We don't need to create session again. We are getting session from the constructor

File: src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+import com.FoodyBuddy.Dao.OrderDishDAO;
+import com.FoodyBuddy.Model.OrderDish;
+
+@Repository
+public class OrderDishDAOImpl implements OrderDishDAO {
+
+	private SessionFactory sessionFactory;
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java]: remove this line

File: src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	public OrderDish getByDishId(Integer DishId) {
+		Session session = this.sessionFactory.openSession();
+		OrderDish orderDish = (OrderDish) session.createQuery("from Order_dish where Order_dish.dish_id =" + DishId).list().get(0);
+		return orderDish;
+	}
+	
+	// returns list of orders by Id
+	@SuppressWarnings("unchecked")
+	public OrderDish getByOrderId(Integer OrderId ) {
+		Session session = this.sessionFactory.openSession();
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java]: same thing here . clean this from each function

File: src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		OrderDish orderDish = (OrderDish) session.load(OrderDish.class, new Integer(id));
+		if(null != orderDish){
+			session.delete(orderDish);
+		}
+		
+	}
+
+	public Integer sumDishPrice(Integer OrderId) {
+		Session session = this.sessionFactory.openSession();
+		int netDishPrice = session.createQuery("SELECT SUM(net_dish_price) FROM order_dish GROUP BY order_id").getFirstResult();
```
Comment (sairamch04) [inline src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java]: Check whether this correct.

File: src/main/java/com/FoodyBuddy/Model/OrderDish.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+@Table(name="order_dish")
+public class OrderDish{
+	
+	@Id
+	@Column(name="id")
+	@GeneratedValue
+	private int id;
+	
+	@Column(name = "order_id", nullable = false)
+    @ManyToOne
```
Comment (kanmaytacker) [inline src/main/java/com/FoodyBuddy/Model/OrderDish.java]: Indentation Error.

File: src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+import com.FoodyBuddy.Dao.OrderDishDAO;
+import com.FoodyBuddy.Model.OrderDish;
+
+@Repository
+public class OrderDishDAOImpl implements OrderDishDAO {
+
+	Session session;
+	
+	public OrderDishDAOImpl(Session session) {
```
Comment (SainathB) [inline src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java]: Indent Constructor properly.

File: src/main/java/com/FoodyBuddy/Model/OrderDish.java [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+	   @Column(name = "updated_at", nullable = false)
+	   private Date updatedAt;
+	    
+	    @Temporal(TemporalType.TIMESTAMP)
+	   @Column(name = "deleted_at", nullable = false)
+	   private Date deletedAt;
+	
+	
+	public OrderDish(int netDishPrice, int quantity, Order order, Dish dish) {
+
```
Comment (SainathB) [inline src/main/java/com/FoodyBuddy/Model/OrderDish.java]: No need to leave empty line here.

File: src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+	}
+	
+	// returns list of orders by Id
+	@SuppressWarnings("unchecked")
+	public OrderDish getByOrderId(Integer OrderId ) {
+		OrderDish orderDish = (OrderDish) session.createQuery("from Order_dish where Order_dish.order_id =" + OrderId).list().get(0);
+		return orderDish;
+	}
+	
+	//
```
Comment (kanmaytacker) [inline src/main/java/com/FoodyBuddy/Dao/Impl/OrderDishDAOImpl.java]: Remove unused comment.

**Comment by junzy** [comment]
is this good to go?

**Comment by sairamch04** [comment]
@junzy  Looks good 👍  We have to discuss on package naming. The standard convention follows all small letters [https://github.com/sairamch04/FoodyBuddy/issues/14]

**Comment by junzy** [comment]
@sairamch04 @kanmaytacker Seems like DAO needs to be dao

**Pull request merged**